src/pint/fitter.py

```diff
@@ -565,10 +565,16 @@ def get_derived_params(self):
                     self.model.TASC.quantity.mjd,
                     self.model.TASC.uncertainty.to(u.d).value,
                 )
-                pb = ufloat(
-                    self.model.PB.quantity.to(u.d).value,
-                    self.model.PB.uncertainty.to(u.d).value,
-                )
+                try:
```

Comment:
Rather than a try with a bare except, a better practice would be something like:
```
if hasattr(self.model, "PB"):
    pb = ...
elif hasattr(self.model, "FB0"):
    p, per = ...
    pb = ...

Reply:
Just made this change and encountered something weird. The test par file I'm using uses an ELL1 binary model with FBn params (PB is not present; same as the par file I was working with where I originally encountered this issue). The test I've written fails with the suggested code because it apparently thinks the model has PB?!

Reply:
OK, it probably also needs `and self.model.PB.value is not None` and the same for `FB0`

Reply:
Yeah, that works (as expected). I'll push the updated version, but I'm still a bit confused why `hasattr(self.model, "PB")` evaluates to True here. (Doesn't really matter for this PR.)

Reply:
depending on how the model is initialized, it might have properties that are there but unset.

Reply:
`hasattr` is fine, but there's nothing wrong with catching exceptions - as long as you catch only the specific exception you are looking for. Just run the code on your test case, see what exception is emitted, and catch that, rather than a bare `except`.